query-parser/tests/find_tests.rs

```diff
@@ -234,3 +235,35 @@ fn can_parse_order_by() {
                Some(vec![Order(Direction::Descending, Variable::from_valid_name("?y")),
                          Order(Direction::Ascending, Variable::from_valid_name("?x"))]));
 }
+
+#[test]
+fn can_parse_limit() {
+    let invalid = "[:find ?x :where [?x :foo/baz ?y] :limit]";
+    assert!(parse_find_string(invalid).is_err());
+
+    let zero_invalid = "[:find ?x :where [?x :foo/baz ?y] :limit 00]";
+    assert!(parse_find_string(zero_invalid).is_err());
+
+    let none = "[:find ?x :where [?x :foo/baz ?y]]";
+    assert_eq!(parse_find_string(none).unwrap().limit,
+               Limit::None);
+
+    let one = "[:find ?x :where [?x :foo/baz ?y] :limit 1]";
+    assert_eq!(parse_find_string(one).unwrap().limit,
+               Limit::Fixed(1));
+
+    let onethousand = "[:find ?x :where [?x :foo/baz ?y] :limit 1000]";
+    assert_eq!(parse_find_string(onethousand).unwrap().limit,
+               Limit::Fixed(1000));
+
+    let variable_with_in = "[:find ?x :in ?limit :where [?x :foo/baz ?y] :limit ?limit]";
+    assert_eq!(parse_find_string(variable_with_in).unwrap().limit,
+               Limit::Variable(Variable::from_valid_name("?limit")));
+
+    let variable_with_in_used = "[:find ?x :in ?limit :where [?x :foo/baz ?limit] :limit ?limit]";
```

Comment:
See previous (long!) comment. Note also that SQLite allows this for both positional and named parameters.